protocols/gossipsub/src/behaviour.rs

```diff
@@ -1608,13 +1639,26 @@ where
         propagation_source: &PeerId,
     ) {
         let fast_message_id = self.config.fast_message_id(&raw_message);
+
         if let Some(fast_message_id) = fast_message_id.as_ref() {
-            if let Some(msg_id) = self.fast_messsage_id_cache.get(fast_message_id) {
+            if let Some(msg_id) = self.fast_message_id_cache.get(fast_message_id) {
                 let msg_id = msg_id.clone();
-                self.message_is_valid(&msg_id, &mut raw_message, propagation_source);
-                if let Some((peer_score, ..)) = &mut self.peer_score {
-                    peer_score.duplicated_message(propagation_source, &msg_id, &raw_message.topic);
+                // Report the duplicate
+                if self.message_is_valid(&msg_id, &mut raw_message, propagation_source) {
```

Comment:
Why just report if the message is valid now?

Reply:
I think this was a bug in the previous code. In principle we shouldn't have invalid messages in the duplicates, and if we do, we penalize them but we don't register it as a duplicate (for scoring). The duplicate would benefit the peer, but we dont want to improve scores for peers that are sending us invalid messages.